### DISCUSSION TITLE
How to create Recurring Job to backup specific volume?

### DISCUSSION THREAD

**Discussion by xorinzor** [longhorn/longhorn#9653] [Q&A] [answered]
I've read this page: https://longhorn.io/docs/1.7.1/snapshots-and-backups/scheduling-backups-and-snapshots/

But even following those steps in the UI it is completely unclear to me how I either assign volumes to a group, or create a recurring job for a specific volume. The resulting manifest does not seem to use any kind of selector.

Can someone clarify this (in the documentation) for me? I don't want to run a backup schedule for every volume. Only very specific ones.
I can make them manually, so there probably is an undocumented way to do it..

EDIT: To clarify, I'm not looking to do this manually in the UI. I would like to write a `RecurringJob` manifest with the options to configure this in longhorn

**Comment by c3y1huang** [comment, marked as answer]
1. Create a RecurringJob [using the manifest](https://longhorn.io/docs/1.7.1/snapshots-and-backups/scheduling-backups-and-snapshots/#using-the-manifest). Replace the group name of your choice:
   ```yaml
   apiVersion: longhorn.io/v1beta1
   kind: RecurringJob
   metadata:
     name: backup-1
     namespace: longhorn-system
   spec:
     cron: "* * * * *"
     task: "backup"
     groups:
     - group1
     retain: 1
     concurrency: 2
   ```
1. After applying the manifest, you can verify the creation of the RecurringJob:
   ```
   > k -n longhorn-system get recurringjobs.longhorn.io 
   NAME       GROUPS       TASK     CRON        RETAIN   CONCURRENCY   AGE   LABELS
   backup-1   ["group1"]   backup   * * * * *   1        2             24s   
   ```
1. Apply the RecurringJob group to a Volume by [labeling the volume to associate it with the recurring job group](https://longhorn.io/docs/1.7.1/snapshots-and-backups/scheduling-backups-and-snapshots/#using-the-kubectl-command).
   ```
   > kubectl -n longhorn-system label volume/<VOLUME-NAME> recurring-job-group.longhorn.io/group1=enabled
   ```
   ```
   > k -n longhorn-system get volume
   NAME                                       DATA ENGINE   STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE            AGE
   pvc-0f8ec752-076e-4bab-8564-29e9e1dccbec   v1            attached   healthy                  21474836480   ip-10-0-2-112   110s
   
   > kubectl -n longhorn-system label volume/pvc-0f8ec752-076e-4bab-8564-29e9e1dccbec recurring-job-group.longhorn.io/group1=enabled
   volume.longhorn.io/pvc-0f8ec752-076e-4bab-8564-29e9e1dccbec labeled
   ```
1. A volume Backup should be created after the scheduled cron time.
   ```
   > k -n longhorn-system get backup
   NAME                      SNAPSHOTNAME                                    SNAPSHOTSIZE   SNAPSHOTCREATEDAT      STATE       LASTSYNCEDAT
   backup-c2f5ece608df4c87   backup-1-d9edffb6-8aa9-486c-a693-ec9f77e56421   522190848      2024-10-17T03:36:06Z   Completed   2024-10-17T03:36:10Z
   ```

**Reply by xorinzor** [reply]
Thanks! Must've missed the step about labeling the volume, didn't realize that's how the schedule is connected to the volume.
